### PR TITLE
chore: node pnp sample for summer

### PR DESCRIPTION
**Pull request by olivakar**

### CODE REVIEW

File: device/samples/pnp/pnpTemperatureController.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  }
+};
+
+const sendCommandResponse = async (request, response, status, payload) => {
+  try {
+    await response.send(status, payload);
+    console.log('Response to method \'' + request.methodName +
+              '\' sent successfully.' );
+  } catch (err) {
+    console.error('An error ocurred when sending a method response:\n' +
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:55]: >\n [](start = 67, length = 2)

Unsure why you put in the newline.

File: device/samples/pnp/pnpTemperatureController.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+              err.toString());
+  }
+};
+
+const helperLogCommandRequest = (request) => {
+  console.log('Received command request for comand name \'' + request.methodName + '\'');
+
+  if (!!(request.payload)) {
+    console.log('The command request paylaod :');
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:64]: > : [](start = 44, length = 2)

Your use of colon in log string is inconsistent with other uses of colon.  (Also, I think there should be a space after the colon but that's an opinion)

File: device/samples/pnp/pnpTemperatureController.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    let patch;
+    if (!!(componentName)) {
+      patch = { };
+      propertiesToReport.__t = "c";
+      patch[componentName] = propertiesToReport;
+    }
+    else {
+      patch = { };
+      patch = propertiesToReport;
+      }
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:80]: >   [](start = 4, length = 2)

Delete spaces.

File: device/samples/pnp/pnpTemperatureController.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+      console.log('The following properties will be updated for root interface:');
+    }
+    console.log(patch);
+    return patch;
+  };
+
+const updateComponentReportedProperties = (deviceTwin, patch, componentName) => {
+  let logLine;
+  if (!!(componentName))
+  {
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:94]: >{ [](start = 2, length = 1)

You generally and the code for others from the Node SDK team have the '{' on the same line as the 'if'

File: device/samples/pnp/pnpTemperatureController.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+    logLine = 'Properties have been reported for component:' + componentName;
+  }
+  else{
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:97]: >else{ [](start = 2, length = 5)

Space needed after the else.  Generally in the node SDK we put the else on the same line as the closing brace for the if.  But you are consistent here so ....

File: device/samples/pnp/pnpTemperatureController.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin) => {
+
+  deviceTwin.on('properties.desired', (delta) => {
+    if (delta[thermostat1ComponentName] || delta[thermostat2ComponentName] || delta[deviceInfoComponentName]) {
+    const version = delta.$version;
+    const componentName = Object.keys(delta)[0];
+    let logLine;
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:112]: Spacing is incorrect.

File: device/samples/pnp/pnpTemperatureController.js [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+  deviceTwin.properties.reported.update(patch, function (err) {
+    if (err) throw err;
+    console.log(logLine);
+  });
+};
+
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin) => {
+
+  deviceTwin.on('properties.desired', (delta) => {
+    if (delta[thermostat1ComponentName] || delta[thermostat2ComponentName] || delta[deviceInfoComponentName]) {
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:109]: > { [](start = 109, length = 2)

Inconsistent use of brace after the if.  (Although I prefer this it's not consistent with your other uses.)

File: device/samples/pnp/pnpTemperatureController.js [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+    let logLine;
+    if (!!(componentName))
+    {
+      logLine = 'Received an update for component: ' + componentName + ' with value: ' + JSON.stringify(delta);
+    }
+    else{
+      logLine = 'Received an update for root interface with value: ' + JSON.stringify(delta);
+    }
+    console.log(logLine);
+    const patch = { [componentName]: {} };
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:121]: >[componentName] [](start = 20, length = 15)

This works out ok for the no componentName case?  patch will be undefined.

File: device/samples/pnp/pnpTemperatureController.js [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+  pnpMsg.contentType = 'application/json';
+  pnpMsg.contentEncoding = 'utf-8';
+  await deviceClient.sendEvent(pnpMsg);
+}
+
+async function sendDatasizeTelemetry(deviceClient, componentName, index) {
+  console.log('Sending telemetry message %d from root interface', index);
+  const data = JSON.stringify({ workingset: 1 + (Math.random() * 90)});
+  const pnpMsg = new Message(data);
+  if (!!(componentName))
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:171]: >if (!!(componentName)) [](start = 2, length = 22)

I'm a bit confused.  Seems as though the code should assert on a component name being defined.  Or  should simply not take a componentName argument.

File: device/samples/pnp/pnpTemperatureController.js [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+    let index1 = 0;
+    let index2 = 0;
+    let index3 = 0;
+    intervalToken1 = setInterval(() => {
+      sendTemperatureTelemetry(client, thermostat1ComponentName, index1).catch((err) => console.log('error', err.toString()));
+      index1 += 1;
+    }, 5000);
+
+    intervalToken2 = setInterval(() => {
+      sendTemperatureTelemetry(client, thermostat2ComponentName, index2).catch((err) => console.log('error', err.toString()));
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:205]: >r' [](start = 105, length = 2)

Seems as though there should be a space in the error string.   Perhaps 'error: '   ?

File: device/samples/pnp/pnpTemperatureController.js [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+    try {
+      resultTwin = await client.getTwin();
+      const patchRoot = helperCreateReportedPropertiesPatch({serialNumber:"alohomora"}, null, false);
+      const patchThermostat1Info = helperCreateReportedPropertiesPatch({
+        targetTemperature: {"value": 56.78, "ac": 200, "ad": "wingardium leviosa", "av": 1},
+        maxTempSinceLastReboot: 67.89,
+      }, thermostat1ComponentName);
+
+      const patchThermostat2Info = helperCreateReportedPropertiesPatch({
+        targetTemperature: {"value": 35.67, "ac": 200, "ad": "expecto patronum", "av": 1},
```
Comment (anthonyvercolano) [inline device/samples/pnp/pnpTemperatureController.js:227]: >expecto patronum" [](start = 62, length = 17)

It's a fun book, but, you MUST change these things.

File: device/samples/pnp/pnpTemperatureController.js (hunk 2 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:64]: removed

File: device/samples/pnp/pnpTemperatureController.js (hunk 3 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:80]: should be fixed now after linting..somehow yesterday the linting did not work correctly and only reported 2 errors for my file

File: device/samples/pnp/pnpTemperatureController.js (hunk 4 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:94]: same reason of linting

File: device/samples/pnp/pnpTemperatureController.js (hunk 5 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:97]: same reason of linting

File: device/samples/pnp/pnpTemperatureController.js (hunk 6 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:112]: same reason of linting

File: device/samples/pnp/pnpTemperatureController.js (hunk 7 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:109]: not sure what u mean here

File: device/samples/pnp/pnpTemperatureController.js (hunk 8 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:121]: code has been changed now...but currently for this method no component will not work. I have a TODO here

File: device/samples/pnp/pnpTemperatureController.js (hunk 10 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:205]: spaced now..but u said colons are not used in node js logs...error is just a higher level of log right ?

File: device/samples/pnp/pnpTemperatureController.js (hunk 11 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js:227]: why oh why ? no one cares... the description can be anything right ?

File: device/samples/pnp/pnpTemperatureController.js (hunk 9 above)
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js]: changed into generic method

File: device/samples/pnp/pnpTemperatureController.js [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+      deviceClient.close();
+      process.exit();
+    } else {
+      console.log('User Input was : ' + data);
+      console.log('Please only enter q or Q to exit sample.');
+    }
+  });
+};
+async function sendTemperatureTelemetry(deviceClient, componentName, index) {
+  console.log('Sending telemetry message %d from component %s ', index, componentName);
```
[the remark this answers is not in the document]
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js]: i don't think this function exists now. both are common fns..i am not sure how are u saying old commit.

File: device/samples/pnp/pnpTemperatureController.js [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+  {
+    pnpMsg.properties.add(messageSubjectProperty, componentName);
+  }
+  pnpMsg.contentType = 'application/json';
+  pnpMsg.contentEncoding = 'utf-8';
+  await deviceClient.sendEvent(pnpMsg);
+}
+
+async function sendDatasizeTelemetry(deviceClient, componentName, index) {
+  console.log('Sending telemetry message %d from root interface', index);
```
[the remark this answers is not in the document]
Comment (olivakar) [reply device/samples/pnp/pnpTemperatureController.js]: https://github.com/olivakar/azure-iot-sdk-node/blob/ok-pnp-summer-2/device/samples/pnp/pnpTemperatureController.js#L145

**Pull request merged**